### DISCUSSION TITLE
Need help to compile and build xDrip with Android Studio

### DISCUSSION THREAD

**Discussion by nasserbek** [NightscoutFoundation/xDrip#2533] [Ideas]
Hello, I'm struggeling to build and compile the last version download from github here 
https://github.com/NightscoutFoundation/xDrip.git

I'm using the last version of Android Studio 2021.3.1 Patch 1.

Is there any step by step procedure for how to compile and build for a non expert with Android Studio.

I'm C and C++, Java programmaer and I need to see the Communication Protocl with the Sensor Libre 2 over Bluetooh.

My goal is to create a Device "ESP32" with Arduino IDE to talk with the Sensor over bluetooth then sends tha Reading to the Clood and alerts over SMS.. From the other side a parent can use his phone with a Blynky app to watch in real time the BG for his kid... The kid will not havea Phone but this small device in his School bag..

All what I need is how to read the Bluetooth Data from Libre 2 in Plain and clear  BG values..

Any help will be appreciated.

Regards

**Comment by tzachi-dar** [comment]
Hi, not answering the question about compilation, as I'm having troubles with it right now.

As for your either idea. I don't think it is as simple as you describe this, since sometimes there is a need to create an encryption key in order to start communication. This also requires NFC capabilities that parcticly exists in a phone.
Your device should either have NFC, and decryption capabilities or be able to talk fast with a device that have this capabilities (a phone running xDrip).
See https://github.com/tzachi-dar/LibreAllHouse/blob/master/LIBRE2.md for rpis that are used for the readings, but remember that this have a fast connection with a phone (wifi not cloud).

Why is the kid having a phone not good enough?

**Reply by steve8x8** [reply]
@tzachi-dar ESP32 boards are a bit smaller than phones these days, I suppose @nasserbek has a very tiny child (and/or doesn't want it to play with the device). Plus, a second approach to the same problem cannot hurt...

@nasserbek There's an app that's supposed (I didn't try) to carry information about a sensor to a phone without NFC. Search for "Juggluco" and its "mirror" feature. (Closed source, and perhaps a bit too reliant on existing libraries that cannot be easily rewritten for, or transformed into, ESP32 code.)

**Reply by nasserbek** [reply]
Yes, my kid is 4 yo and cann't use the phone or even leave it in his school bag.
I've used the TTGO Exp32 in many IoT applications, it is very tiny and has BLE, WIFI and GPRS I can also ad a BM019 NFC to Serial card to have fast scanning but I don't want to add more objects on his arm as with the LimiTTEr NFC to Bluetooth project which is easy to realize with Arduino.

I know the data from sensor are encrypted and I'm able to read them every 1 minute but I'm not able to decrypt them to get the BG values and sendo to the Cloud and this is my main problem,

This why I'm looking for any source to retreive this part of the APP and use it, I know that rewriting Libraries not such easy but I don't need them becuase the ESP32 has its own BLE Libraries and I need only the Communication protocol , hash table and key decryption.

Juggluco as I know is not open source , either Diabox .

The PI project is nice but doesn't fit my scope...

I thought xDrip+ bieng open source it would be easy to understand how the communication with the sensor works but it is really difficult to get it compiled or built in Android Studio..

Regards

**Comment by tzachi-dar** [comment]
If you are using limitter or similar commercial device (I would use some version of the miomio, but it does not realy matter) than your situation is much simpler. What you need to do is write things to the cloud and then xDrip will be able to read it (without changes). This kind of changes do not require encryption at the client side.
xDrip knows today how to read from this repository: https://github.com/tzachi-dar/node-js-getting-started, so you just need to add the code that writes to there.

As for compiling, try gradlew assembledebug from command line, and if this does not work, try installing android studio bumblebee 2021.1.1 patch 1

**Comment by tzachi-dar** [comment]
Here is what I do to compile from command line:
Make sure that the file local.properties points to your sdk.
In my case it looks like:
sdk.dir=C\:\\Users\\nirit\\AppData\\Local\\Android\\Sdk

Make sure you are using java 11.
Here are how versions look in my case:
C:\Users\nirit\take4\xDrip>gradlew --version

------------------------------------------------------------
Gradle 6.8
------------------------------------------------------------

Build time:   2021-01-08 16:38:46 UTC
Revision:     b7e82460c5373e194fb478a998c4fcfe7da53a7e

Kotlin:       1.4.20
Groovy:       2.5.12
Ant:          Apache Ant(TM) version 1.10.9 compiled on September 27 2020
JVM:          11.0.16.1 (Oracle Corporation 11.0.16.1+1-LTS-1)
OS:           Windows 8.1 6.3 amd64

Than run:
gradlew assembleproddebug

**Reply by nasserbek** [reply]
Great, worked from the first time following your instructions with Android Studio bumblebee 2021.1.1 patch 1.. I', able to Build and run the App on the Vitrual Phone.... I really appreciate your help ...
